### PR TITLE
Enforce auth

### PR DESCRIPTION
**Pull request by allthesignals**
This PR enforces auth by throwing an exception when `contactid` is specifically not found. Before, it was only checking if `session` was visible.

This also has the browser display error messages.

### CODE REVIEW

File: server/src/app.controller.ts [hunk 1]
```diff
@@ -36,6 +36,12 @@ export class AppController {
     }
   }
 
+  @Get('/logout')
+  async logout(@Res() res: Response) {
+    res.clearCookie('token')
+      .send({ message: 'Login successful!' });
```
Comment (allthesignals) [inline server/src/app.controller.ts:42]: This is the wrong message...

Comment (bfreeds) [inline server/src/app.controller.ts:42]: Note:  update this message

File: server/src/http-exception.filter.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import { Request, Response } from 'express';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  catch(exception: HttpException, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+    const status = exception.getStatus();
+
+    response
```
Comment (bfreeds) [inline server/src/http-exception.filter.ts:11]: Made to fit json api spec:   https://jsonapi.org/format/#errors

File: server/src/http-exception.filter.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { ExceptionFilter, Catch, ArgumentsHost, HttpException } from '@nestjs/common';
+import { Request, Response } from 'express';
+
+@Catch(HttpException)
```
Comment (bfreeds) [inline server/src/http-exception.filter.ts:4]: Note:  here's the docs on nest exception filters & error handling.
https://docs.nestjs.com/exception-filters#exception-filters

**Pull request merged**